runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java

```diff
@@ -168,15 +197,126 @@ else if (config.clue() && message.toLowerCase().startsWith("!clues") && message.
 			log.debug("Running clue lookup for {}", search);
 			executor.submit(() -> playerClueLookup(setMessage.getType(), setMessage, search));
 		}
+		else if (message.toLowerCase().startsWith("!kc "))
+		{
+			String search = message.substring(4);
+
+			log.debug("Running killcount lookup for {}", search);
+			executor.submit(() -> killCountLookup(setMessage.getType(), setMessage, search));
+		}
+
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage chatMessage)
+	{
+		if (chatMessage.getType() != ChatMessageType.SERVER && chatMessage.getType() != ChatMessageType.FILTERED)
+		{
+			return;
+		}
+
+		String message = chatMessage.getMessage();
+		Matcher matcher = KILLCOUNT_PATERN.matcher(message);
+		if (matcher.find())
+		{
+			String boss = matcher.group(1);
+			int kc = Integer.parseInt(matcher.group(2));
+
+			setKc(boss, kc);
+		}
+
+		matcher = WINTERTODT_PATERN.matcher(message);
+		if (matcher.find())
+		{
+			int kc = Integer.parseInt(matcher.group(1));
+
+			setKc("Wintertodt", kc);
+		}
+	}
+
+	@Subscribe
+	public void onChatboxInput(ChatboxInput chatboxInput)
+	{
+		String value = chatboxInput.getValue();
+		if (!value.startsWith("!kc "))
+		{
+			return;
+		}
+
+		chatboxInput.setStop(true);
+
+		String boss = value.substring(4);
+
+		int kc = getKc(boss);
+		if (kc <= 0)
+		{
+			chatMessageManager.queue(QueuedMessage.builder()
+				.type(ChatMessageType.GAME)
+				.runeLiteFormattedMessage("Unknown kill count for " + boss)
+				.build());
+			return;
+		}
+
+		executor.execute(() ->
+		{
+			try
+			{
+				killCountClient.submit(client.getLocalPlayer().getName(), boss, kc);
+
+				commandManager.sendChatboxInput(chatboxInput.getChatType(), chatboxInput.getValue());
+			}
+			catch (Exception ex)
+			{
+				log.warn("unable to submit killcount", ex);
+			}
+		});
+	}
+
+	private void killCountLookup(ChatMessageType type, SetMessage setMessage, String search)
+	{
+		String player;
+		if (type.equals(ChatMessageType.PRIVATE_MESSAGE_SENT))
+		{
+			player = client.getLocalPlayer().getName();
+		}
+		else
+		{
+			player = sanitize(setMessage.getName());
+		}
+
+		int kc;
+		try
+		{
+			kc = killCountClient.get(player, search);
```

Comment:
The UX of this means that if an `IOException` occurs the user gets no feedback without checking their logs - Could we not present the user with a nicely formatted "Something went wrong" message?